client/app/widgets/QuickCreate/_tests_/QuickCreate.spec.jsx

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { QuickCreate } from '../index';
+
+const checkboxes = [
+  { id: 'last', value: 'a_value', label: 'zoo_label' },
+  { id: 'middle', value: 'middle_value', label: 'middle_label' },
+  { id: 'first', value: 'z_value', label: 'alpha_label' },
+  { id: 'dm', value: 'duplicate_middle_value', label: 'middle_label' },
+];
+
+describe('QuickCreate', () => {
+  let wrapper;
+  beforeEach(() => {
+    wrapper = mount(
+      <QuickCreate
+        name="name"
+        id="123"
+        label="label"
+        checkboxes={checkboxes}
+        formProps={{ inputs: [{}] }}
+      />,
+    );
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+    wrapper.unmount();
+  });
+
+  it('sets correct initial values in state', () => {
+    const state = wrapper.state();
+    expect(state).toHaveProperty('checkboxes');
+    expect(state).toHaveProperty('open', false);
+    expect(state).toHaveProperty('accordionOpen', false);
+  });
+
+  it('sorts checkboxes by label', () => {
+    const checkboxesFromState = wrapper.state('checkboxes');
+    expect(checkboxesFromState[0].id).toBe('first');
+    expect(checkboxesFromState[3].id).toBe('last');
+  });
+
+  it('renders Modal initially closed', () => {
+    const modal = wrapper.find('Modal');
+    expect(modal).toHaveLength(1);
+    expect(modal.prop('open')).toBeFalsy();
+  });
+
+  describe('when input is changed', () => {
+    function changeInput(label) {
+      expect(wrapper.find('DynamicForm')).toHaveLength(0);
+      expect(wrapper.find('Modal').prop('open')).toBeFalsy();
+      wrapper.find('Input').prop('onChange')({
+        checkboxes,
+        label,
+      });
+      wrapper.update();
+    }
+
+    it('opens modal and displays quick create form if label does not exist', () => {
+      changeInput('new_label');
+      expect(wrapper.find('Modal').prop('open')).toBeTruthy();
+      expect(wrapper.find('DynamicForm')).toHaveLength(1);
+    });
+
+    it('does nothing if label exists', () => {
```

Comment:
I would rephrase this to, 'does not open the modal if the label already exists'.